src/main/java/menion/android/whereyougo/network/LoginTask.java

```diff
@@ -0,0 +1,237 @@
+                loginCheckdialog.setMessage("Something unexpected happened. Please close the dialog!");
+                break;
+        }
+    }
+
+    @Override
+    protected void onCancelled() {
+        super.onCancelled();
+        loginCheckdialog.dismiss();
+    }
+
+    private boolean init() {
+        try {
+            System.setProperty("http.keepAlive", "false");
+            httpClient = new OkHttpClient.Builder()
+                .sslSocketFactory(new TLSSocketFactory())
+                .cookieJar(new NonPersistentCookieJar())
+                .connectTimeout(30, TimeUnit.SECONDS)
+                .readTimeout(30, TimeUnit.SECONDS)
+                .writeTimeout(30, TimeUnit.SECONDS)
+                .build();
+        } catch (KeyManagementException | NoSuchAlgorithmException e) {
+            Logger.e(TAG, "init()", e);
+            errorMessage = e.getMessage();
+        }
+        if (httpClient == null) {
+            publishProgress(new Progress(Task.INIT, State.FAIL, errorMessage));
+        }
+        return httpClient != null;
+    }
+
+    private boolean ping() {
+        Request request = new Request.Builder()
+            .url(LOGIN)
+            .build();
+        return handleRequest(request, Task.PING) != null;
+    }
+
+    private boolean login() {
+        RequestBody formBody = new FormBody.Builder()
+            .add("__EVENTTARGET", "")
+            .add("__EVENTARGUMENT", "")
+            .add("ctl00$ContentPlaceHolder1$Login1$Login1$UserName", username)
+            .add("ctl00$ContentPlaceHolder1$Login1$Login1$Password", password)
+            .add("ctl00$ContentPlaceHolder1$Login1$Login1$LoginButton", "Sign In")
+            .build();
+        Request request = new Request.Builder()
+            .url(LOGIN)
+            .post(formBody)
+            .build();
+        publishProgress(new Progress(Task.LOGIN, State.WORKING));
+        Response response = handleRequest(request);
+        if (response != null && !LOGIN.equals(response.request().url().toString())) {
+            publishProgress(new Progress(Task.LOGIN, State.SUCCESS));
+            return true;
+        } else {
+            publishProgress(new Progress(Task.LOGIN, State.FAIL, errorMessage));
+            return false;
+        }
+    }
+
+    private boolean logout() {
+        RequestBody formBody = new FormBody.Builder()
+            .add("__EVENTTARGET", "ctl00$ProfileWidget$LoginStatus1$ctl00")
+            .add("__EVENTARGUMENT", "")
+            .build();
+        Request request = new Request.Builder()
+            .url(LOGIN)
+            .post(formBody)
+            .build();
+        return handleRequest(request, Task.LOGOUT) != null;
+    }
+
+    private Response handleRequest(Request request, Task task) {
+        publishProgress(new Progress(task, State.WORKING));
+        Response response = handleRequest(request);
+        if (response != null) {
+            publishProgress(new Progress(task, State.SUCCESS));
```

Comment:
The State.SUCCESS should only be set after the last task (Task.LOGOUT), on the other stay at State.WORKING. 
Then the change in Line [76] to "close dialog" is only in this moment.